<?xml version="1.0"?>
<launch>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="paused" value="false"/>
    <arg name="gui" value="true"/> 
  </include>

  <!-- diffdrive -->
  <group ns="mim_diffdrive_arm">
    <arg name="diffdrive_init_pose" value="-x 0 -y -1 -z 0 -R 0 -P 0 -Y 0"/>
    <param name="robot_description" command="$(find xacro)/xacro $(find mim_robots)/urdf/diffdrive_arm.xacro"/>
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_model_diffdrive" args="$(arg diffdrive_init_pose) -param robot_description -urdf -model mim_diffdrive_arm" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen" >
      <param name="publish_frequency" value="40.0"/>
    </node>

    <rosparam command="load" 
      file="$(find mim_robots)/config/diffdrive.yaml"/>
    <rosparam command="load" 
      file="$(find mim_robots)/config/arm.yaml"/>

    <node name="diffdrive_controller_manager" pkg="controller_manager" type="spawner"
      respawn="false" output="screen" 
      args="mim_diffdrive/mobile_base_controller
            mim_arm/joint1_controller 
            mim_arm/joint2_controller 
            mim_arm/joint3_controller 
            mim_arm/joint4_controller 
            mim_arm/joint5_controller 
            mim_arm/joint6_controller 
            mim_arm/gripper1_controller
            mim_arm/gripper2_controller" />
  </group>

  <!-- rocker_boggie -->
  <group ns="mim_rockerb_arm">
    <arg name="init_pose" value="-x 0 -y 1 -z 0 -R 0 -P 0 -Y 0"/>
    <param name="robot_description" command="$(find xacro)/xacro $(find mim_robots)/urdf/rockerb_arm.xacro"/>
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_model_rockerb" args="$(arg init_pose) -param robot_description -urdf -model mim_rockerb_arm"/> 
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen" >
      <param name="publish_frequency" value="40.0"/>
    </node>

    <rosparam command="load"
      file="$(find mim_robots)/config/rockerb.yaml"/>
    <rosparam command="load" 
      file="$(find mim_robots)/config/arm.yaml"/>

    <node name="rockerb_controller_manager" pkg="controller_manager" type="spawner"
      respawn="false" output="screen"  
      args="mim_rockerb/mobile_base_controller
            mim_arm/joint1_controller 
            mim_arm/joint2_controller 
            mim_arm/joint3_controller 
            mim_arm/joint4_controller 
            mim_arm/joint5_controller 
            mim_arm/joint6_controller 
            mim_arm/gripper1_controller
            mim_arm/gripper2_controller" />
  </group>

  <!-- ramp -->
  <arg name="ramp1_init_pose" value="-x 1 -y 1.5 -z 0 -R 0 -P 0 -Y 0"/>
  <param name="ramp1_description" command="$(find xacro)/xacro $(find mim_robots)/urdf/other_models/ramp.urdf"/>

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_model_ramp1" args="$(arg ramp1_init_pose) -param ramp1_description -urdf -model ramp1"/> 

  <!-- ramp -->
  <arg name="ramp2_init_pose" value="-x 1 -y -0.5 -z 0 -R 0 -P 0 -Y 0"/>
  <param name="ramp2_description" command="$(find xacro)/xacro $(find mim_robots)/urdf/other_models/ramp.urdf"/>

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_model_ramp2" args="$(arg ramp2_init_pose) -param ramp2_description -urdf -model ramp2"/> 

</launch>
